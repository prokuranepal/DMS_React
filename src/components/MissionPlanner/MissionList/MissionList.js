import React from 'react';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import { Typography, Paper } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import {Link} from 'react-router-dom'
const useStyles = makeStyles((theme) => ({
    root: {
        width: '90%',

        margin: '10px auto',
    },
    button: {
        margin: 'auto'
    },
    data: {
        margin: 'auto',
        marginLeft: '20px'
    },
    paper: {

        width: '90%',
        backgroundColor: '#5C9BDA',
        color: 'white',
        margin: `${theme.spacing(0.2)}px auto`,
        padding: theme.spacing(2),
    },
}));
const MissionList = props => {
    const classes = useStyles();
    return (
        <Grid container className={classes.root}>

            <Paper className={classes.paper}>
                <Grid container wrap="nowrap" spacing={2}>
                    <Grid item xs={12} sm={8} md={10}>
                        <Typography>Mission ID: {props.id}</Typography>
                        <Typography>Source: {props.source}</Typography>
                        <Typography>Destination: {props.destination}</Typography>
                    </Grid>
                    <Grid container item xs={0} sm={4} md={2}>
                    <Link className={classes.button}to='/admin/missionplanner/missionview' style={{textDecoration:'none', color: 'white'}}>
                        <Button variant="contained" color='secondary' >
                            View
                </Button>
                </Link>
                    </Grid>
                </Grid>
            </Paper>
            {/* </Paper> */}
        </Grid>)
}

export default MissionList;