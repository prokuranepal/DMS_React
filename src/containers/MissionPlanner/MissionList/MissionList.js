import React from 'react';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import { Typography, Paper } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import {Link} from 'react-router-dom';
const useStyles = makeStyles((theme) => ({
    root: {
        width: '90%',

        margin: '10px auto',
    },
    button: {
        margin: 'auto'
    },
    data: {
        margin: 'auto',
        marginLeft: '20px'
    },
    paper: {

        width: '90%',
        backgroundColor: '#5C9BDA',
        color: 'white',
        margin: `${theme.spacing(0.2)}px auto`,
        padding: theme.spacing(2),
    },
}));

/**
* A single drone in the drone list containing a label with a radio
*
* @param {AlphaNumeric} props.id - a unique ID of a mission
* @param {string} props.source - take-off place of the drone
* @param {Integer} props.destination - payload drop place of the drone
* @returns {MissionList} - It returns mission information with a view button
*/


const MissionList = props => {
    const classes = useStyles();
    return (
        <Grid container className={classes.root}>

            <Paper className={classes.paper}>
                <Grid container wrap="nowrap" spacing={2}>
                    <Grid item xs={12} sm={8} md={10}>
                        <Typography data-test="id-component">Mission ID: {props.id}</Typography>
                        <Typography data-test="source-component">Source: {props.source}</Typography>
                        <Typography data-test="destination-component">Destination: {props.destination}</Typography>
                    </Grid>
                    <Grid container item xs={0} sm={4} md={2}>
                    <Link className={classes.button} to='/app/missionplanner/missionview' style={{textDecoration:'none', color: 'white'}}>
                        <Button variant="contained" color='secondary' >
                            View
                </Button>
                </Link>
                    </Grid>
                </Grid>
            </Paper>
            {/* </Paper> */}
        </Grid>)
}

export default MissionList;