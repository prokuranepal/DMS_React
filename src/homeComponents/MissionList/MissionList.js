import React, {useEffect} from 'react';
import {useSelector, useDispatch} from 'react-redux';
import MissionListItem from './MissionListItem/MissionListItem';
import CustomScrollbars from '../../util/CustomScrollbars';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormControl from '@material-ui/core/FormControl';
import FormLabel from '@material-ui/core/FormLabel';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Spinner from '../UI/Spinner/Spinner';
import * as actions from '../../store/actions/droneControl';

const useStyles = makeStyles((theme) => ({
    header: {
        fontSize: '20px',
        color: 'white',
        margin: 'auto',
        padding: '10px'
    },
    buttonLayout: {
        display: 'flex',
        flexDirection: 'row',
        margin: '20px 20px 15px',
        padding: '0px 30px'
    },
    button: {
        alignSelf: 'flex-end',
        marginLeft: 'auto'
    }

}));

/**
* A List of missions to be selected to upload to a drone
*
* @param {Function} props.abort - onclick function to abort the selection of mission.
* @param {Function} props.select - function to select a mission
* @returns {MissionList} - returns MissionListItem and buttons
*/
const MissionList = (props) => {
    const classes = useStyles();
    const dispatch = useDispatch();
    const [value, setValue] = React.useState('1');
    
    const handleChange = (event) => {
        setValue(event.target.value);
    };
    const missions = useSelector(({ droneControl }) => droneControl.missions);

    useEffect(() => {
        dispatch(actions.fetchMissionList());
    },[]);
    return (
        // <div className="module-list mail-list">
       <div>
           {missions !== null?
        <FormControl component="fieldset">
            <p className={classes.header}>Select Mission</p>
        <CustomScrollbars className="module-list-scroll scrollbar"
            style={{ height: 200 >= 800 ? 'calc(100vh - 600px)' : 'calc(100vh - 400px)', minWidth: '300px' }}>
            <RadioGroup data-test="radioGroup" aria-label="gender" name="gender1" value={value} onChange={handleChange}>
                {missions.map((mission, index) =>
                    // <FormControlLabel control={}>
                    <MissionListItem key={index} mission={mission} data-test="missionList" />
                )}
            </RadioGroup>
        </CustomScrollbars>
        <div className={classes.buttonLayout}>
                        <Button variant="outlined" color='primary' onClick={props.abort} data-test="abortButton">Abort</Button>
                        <Button variant="outlined" color="primary" className={classes.button} onClick={() => props.select(value)} data-test="selectButton">Select</Button>
                        </div>
        </FormControl>:<Spinner/>}
        </div>
        
    )
};

export default MissionList;