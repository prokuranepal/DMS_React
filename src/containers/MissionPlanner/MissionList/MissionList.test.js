import React from 'react';
import MissionList from './MissionList';
import {
    configure,
    shallow
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})


const setup = (props = {}, state = null) => {
    return (shallow( < MissionList {...props}  />).dive()
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}


const dummy_data={id:"1",source:'Dharan',destination:"biratnagar"}
describe('MissionList />', () => {
  let wrapper;


  beforeEach(() => {
    wrapper = setup(dummy_data)
  });

  afterEach(() => {
    jest.clearAllMocks();
  });


it("MissionList snapshot", () => {
        // let wrapper = setup(dummy_data);
        expect(wrapper).toMatchSnapshot();

  
})

it("MissionList Components", () => {

   const idComponent = findByTestAttr(wrapper, "id-component")
   const sourceComponent = findByTestAttr(wrapper, "source-component")
   const destinationComponent = findByTestAttr(wrapper, "destination-component")

    expect(idComponent.text()).toEqual("Mission ID: 1");
    expect(sourceComponent.text()).toEqual("Source: Dharan");
    expect(destinationComponent.text()).toEqual("Destination: biratnagar");

  


})
})
