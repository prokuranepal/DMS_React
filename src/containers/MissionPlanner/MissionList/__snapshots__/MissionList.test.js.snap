// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionList /> MissionList snapshot 1`] = `ShallowWrapper {}`;
